torch/_inductor/codegen/simd.py

```diff
@@ -1280,13 +1282,30 @@ def codegen_node_schedule(
             isinstance(node, BaseSchedulerNode) and node.is_split_scan()
             for node in node_schedule
         )
+
         kernel_type = TritonSplitScanKernel if is_split_scan else self.kernel_type
         kernel_args = tiled_groups
-        kernel_kwargs = {
-            "reduction_hint": reduction_hint_val,
-            "mutations": mutations,
-            "index_dtype": index_dtype,
-        }
+        kernel_kwargs = dict(
+            reduction_hint=reduction_hint_val,
+            mutations=mutations,
+            index_dtype=index_dtype,
+        )
+
+        def _node_has_sort(node):
+            if node in (EnableReduction, DisableReduction):
+                return False
+
+            sort_nodes = node._body.root_block.graph.find_nodes(
+                op="call_method", target="sort"
+            )
+            return bool(sort_nodes)
+
+        # ops.sort only works with persistent reduction, and is not bandwidth bound anyway
+        # so taking the hit of non-coalesced loads is okay
+        has_sort = any(_node_has_sort(node) for node in node_schedule)
```

Comment:
This is a bit of a work-around. For outer reductions we don't normally use `RBLOCK >= 64` but for sort since we can't loop over we really need the longer `RBLOCK` as we don't have a non-persistent version of the sort. So, I just add an exception if I find `ops.sort` in the kernel.